packages/pmml-editor/src/editor/components/LinearRegressionViewer/molecules/LinearRegressionViewAdaptor.tsx

```diff
@@ -0,0 +1,182 @@
+  PMML,
+  RegressionModel,
+  RegressionTable
+} from "@kogito-tooling/pmml-editor-marshaller";
+import { Line, LinearRegressionView, Range } from "./LinearRegressionView";
+import { useSelector } from "react-redux";
+
+interface LinearRegressionViewAdaptorProps {
+  model: RegressionModel;
+}
+
+const LinearRegressionViewAdaptor = (props: LinearRegressionViewAdaptorProps) => {
+  const { model } = props;
+
+  const dataDictionary = useSelector<PMML, DataDictionary>((state: PMML) => state.DataDictionary);
+
+  const table: RegressionTable | undefined = getRegressionTable(model);
+  if (table === undefined) {
+    return <div>Unsupported</div>;
+  }
+
+  const numericPredictor: NumericPredictor | undefined = getNumericPredictorType(table);
+  if (numericPredictor === undefined) {
+    return <div>Unsupported</div>;
+  }
+
+  const modelName: string | undefined = model.modelName;
+  const miningSchema: MiningSchema = model.MiningSchema;
+  const dependentAxisTitle = miningSchema.MiningField.filter(mf => mf.usageType === "target")[0].name;
+  const lines: Line[] = getLines(table, numericPredictor);
+
+  //Get Ranges from DataDictionary or use reasonable defaults
+  let rangeY: Range | undefined = getYRange(dataDictionary, model);
+  if (rangeY === undefined) {
+    rangeY = getDefaultYRange(lines);
+  }
+  let rangeX: Range | undefined = getXRange(dataDictionary, numericPredictor);
+  if (rangeX === undefined) {
+    rangeX = getDefaultXRange(lines, rangeY);
+  }
+
+  return (
+    <div>
+      <LinearRegressionView
+        modelName={modelName ?? "<Undefined>"}
+        independentAxisTitle={numericPredictor.name as string}
+        dependentAxisTitle={dependentAxisTitle as string}
+        lines={lines}
+        rangeX={rangeX}
+        rangeY={rangeY}
+      />
+    </div>
+  );
+};
+
+function getRegressionTable(model: RegressionModel): RegressionTable | undefined {
+  const tables: RegressionTable[] = model.RegressionTable;
+  if (tables === undefined || tables.length > 1) {
+    return undefined;
+  }
+  return tables[0];
+}
+
+function getNumericPredictorType(table: RegressionTable): NumericPredictor | undefined {
+  const predicators: NumericPredictor[] | undefined = table.NumericPredictor;
+  if (predicators === undefined || predicators.length > 1) {
+    return undefined;
+  }
+  return predicators[0];
+}
+
+function getLines(table: RegressionTable, numbericPredictor: NumericPredictor): Line[] {
+  const c: number = table.intercept;
+  const line: Line = { m: numbericPredictor.coefficient, c: c, title: "base" };
+  const lines: Line[] = new Array<Line>(line);
+
+  //We need to duplicate the line for each CategoricalPredictor
+  const categoricalPredictors: CategoricalPredictor[] | undefined = table.CategoricalPredictor;
+  if (categoricalPredictors === undefined) {
+    return lines;
+  }
+
+  categoricalPredictors.forEach(cp => {
+    lines.push({ m: line.m, c: line.c + cp.coefficient, title: line.title + " (" + cp.value + ")" });
```

Comment:
Would template literals make the last argument more readable?
```suggestion
    lines.push({ m: line.m, c: line.c + cp.coefficient, title: `${line.title} (${cp.value})` });
```

Reply:
Applied in new commit. Thank-you @jstastny-cz.